Doc/library/codecs.rst

```diff
@@ -414,42 +438,58 @@ functions:
 
 .. function:: strict_errors(exception)
 
-   Implements the ``'strict'`` error handling: each encoding or
-   decoding error raises a :exc:`UnicodeError`.
+   Implements the ``'strict'`` error handling.
 
+   Each encoding or decoding error raises a :exc:`UnicodeError`.
 
-.. function:: replace_errors(exception)
 
-   Implements the ``'replace'`` error handling (for :term:`text encodings
-   <text encoding>` only): substitutes ``'?'`` for encoding errors
-   (to be encoded by the codec), and ``'\ufffd'`` (the Unicode replacement
-   character) for decoding errors.
+.. function:: ignore_errors(exception)
 
+   Implements the ``'ignore'`` error handling.
 
-.. function:: ignore_errors(exception)
+   Malformed data is ignored and encoding or decoding is continued without
+   further notice.
 
-   Implements the ``'ignore'`` error handling: malformed data is ignored and
-   encoding or decoding is continued without further notice.
 
+.. function:: replace_errors(exception)
 
-.. function:: xmlcharrefreplace_errors(exception)
+   Implements the ``'replace'`` error handling.
 
-   Implements the ``'xmlcharrefreplace'`` error handling (for encoding with
-   :term:`text encodings <text encoding>` only): the
-   unencodable character is replaced by an appropriate XML character reference.
+   Substitutes ``?`` (ASCII character) for encoding errors, or ``U+FFFD`` (the
+   official REPLACEMENT CHARACTER) for decoding errors.
 
 
 .. function:: backslashreplace_errors(exception)
 
-   Implements the ``'backslashreplace'`` error handling (for
-   :term:`text encodings <text encoding>` only): malformed data is
-   replaced by a backslashed escape sequence.
+   Implements the ``'backslashreplace'`` error handling.
+
+   Malformed data is replaced by a backslashed escape sequence.
+   On encoding, use hexadecimal form of Unicode code point with formats
+   ``\xhh`` ``\uxxxx`` ``\Uxxxxxxxx``. On decoding, use hexadecimal form of
+   byte value with format ``\xhh``.
+
+   .. versionchanged:: 3.5
+      Now works with decoding and translating.
+
+
+.. function:: xmlcharrefreplace_errors(exception)
+
+   Implements the ``'xmlcharrefreplace'`` error handling (for encoding within
+   :term:`text encoding` only).
+
+   The unencodable character is replaced by an appropriate XML/HTML numeric
+   character reference, which is a decimal form of Unicode code point with
+   format ``&#num;``
+
 
 .. function:: namereplace_errors(exception)
 
-   Implements the ``'namereplace'`` error handling (for encoding with
-   :term:`text encodings <text encoding>` only): the
-   unencodable character is replaced by a ``\N{...}`` escape sequence.
+   Implements the ``'namereplace'`` error handling (for encoding within
+   :term:`text encoding` only).
+
+   The unencodable character is replaced by a ``\N{...}`` escape sequence,
+   what appears in the brace is the Name property from Unicode Character
```

Comment:
```suggestion
   The sequence that appears in the brace is the *name* property from Unicode Character
```

Reply:
I'm afraid the reader may confuse, what is the "sequence" refered to in this paragraph?
`\N{LATIN SMALL LETTER SHARP S}`
or
`LATIN SMALL LETTER SHARP S`

Reply:
@animalize 
> I'm afraid the reader may confuse, what is the "sequence" refered to in this paragraph?

Hmm, I see your point, what do you think of "The sequence of characters that appear within the braces ..."? This would differentiate it from the escape sequence without being as vague as the original "what appears" .

Reply:
I would like to avoid using "sequence" again, any idea?

Reply:
Perhaps "The set of characters ..."?

Reply:
Also note that "brace" should be "braces", since it is enclosed within a pair of braces. I did not include that in my initial suggestion.

Reply:
I'd keep ``Name``, usually ``*name*`` style is used for function parameter in Python documentation.

Add an example to this item.
After adding this example, any advice for "what appears"?
![](https://raw.githubusercontent.com/animalize/pics/master/issues/nameeh.png)

The above table also has these contents, but I hope the sentences in the table keep short.

Reply:
@animalize I was suggesting for "The set of characters" to replace "what appears".
 
Here are all of the changes I would suggest for the second paragraph:
```
All unencodeable characters are replaced with a ``\N{..}`` escape sequence. The set of
characters that appear in the braces represent the ``Name`` property from the Unicode
Character Database. For example, the German lowercase letter ... will be converted to
the byte sequence ...
```
Note: Includes a few new suggestions and doesn't include the line wrapping.

Reply:
To be honest, I want the documentation to be simple. 
When people are busy with task, they can get the knowledge at a glance.

Long texts may be more standardized, but add a little cost of understanding.
Maybe the tutorial part needs a more standard language, but the reference part can be simple a bit.

Reply:
@animalize:
> To be honest, I want the documentation to be simple.

Is there a particular part of the recommendation that you thought was not phrased in a simple manner? If so, I can provide an alternative with only the grammar corrections. I'd be glad to further work on improving it, that was just a first idea. 

>  Maybe the tutorial part needs a more standard language, but the reference part can be simple a bit.

Based on my experience from the documentation changes and reviews I've made so far, it seems to be the other way around for the most part.

The tutorial tends to be the less formalized part and more simple, while the language reference focuses more on technical accuracy. @JulienPalard and @willingc might be able to provide more accurate information on that though (since they're both documentation experts).

Reply:
I mean "The set of characters" is a little longer than "what appears", but this is only a minor problem. 😉

I adopted your suggestion in the commit.